File: backend/src/main/java/boukevanzon/Anchiano/model/Workspace.java
package boukevanzon.Anchiano.model;

import jakarta.persistence.*;
import lombok.*;
import java.time.LocalDateTime;

@Entity
@Table(name = "workspaces")
@Data
@NoArgsConstructor
@Getter
@Setter
@AllArgsConstructor
@Builder

public class Workspace {
    @Id @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false, length = 150)
    private String name;

    @Column(length = 1000)      
    private String description;


    @ManyToOne(optional = false)
    @JoinColumn(name = "owner_id")
    private User owner;

    @Builder.Default
    @Column(nullable = false)
    private LocalDateTime createdAt = LocalDateTime.now();

    @Builder.Default
    @Column(nullable = false)
    private LocalDateTime updatedAt = LocalDateTime.now();

    @Column(nullable = true)
    private LocalDateTime deletedAt;
}

File: frontend/lib/features/workspaces/models/workspace.dart
import '../../auth/models/user.dart';

class Workspace {
  final int id;
  final String name;
  final String description;
  final User owner;
  final DateTime createdAt;
  final DateTime updatedAt;
  final DateTime? deletedAt;

  Workspace({
    required this.id,
    required this.name,
    required this.description,
    required this.owner,
    required this.createdAt,
    required this.updatedAt,
    this.deletedAt,
  });

  factory Workspace.fromJson(Map<String, dynamic> json) {
    return Workspace(
      id: json['id'],
      name: json['name'],
      description: json['description'],
      owner: User.fromJson(json['owner']),
      createdAt: DateTime.parse(json['createdAt']),
      updatedAt: DateTime.parse(json['updatedAt']),
      deletedAt:
          json['deletedAt'] != null ? DateTime.parse(json['deletedAt']) : null,
    );
  }
}
